macros/context_existance_check.sql

```diff
@@ -0,0 +1,163 @@
+      ],
+      "snowplow__enable_yauaa": [
+          'contexts_nl_basjes_yauaa_context_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__yauaa_context')
+      ],
+      "snowplow__enable_ua": [
+          'contexts_com_snowplowanalytics_snowplow_ua_parser_context_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__ua_parser_context')
+      ],
+      "snowplow__enable_application_context": [
+          'contexts_com_snowplowanalytics_mobile_application_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__application_context')
+      ],
+      "snowplow__enable_browser_context": [
+          'contexts_com_snowplowanalytics_snowplow_web_page_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__browser_context')
+      ],
+      "snowplow__enable_geolocation_context": [
+          'contexts_com_snowplowanalytics_snowplow_geolocation_context_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__geolocation_context')
+      ],
+      "snowplow__enable_screen_context": [
+          'contexts_com_snowplowanalytics_mobile_screen_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__screen_context')
+      ],
+      "snowplow__enable_deep_link_context": [
+          'contexts_com_snowplowanalytics_mobile_deep_link_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__deep_link_context')
+      ],
+      "snowplow__enable_screen_summary_context": [
+          'contexts_com_snowplowanalytics_mobile_screen_summary_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__screen_summary_context')
+      ],
+      "snowplow__enable_consent": [
+          'unstruct_event_com_snowplowanalytics_snowplow_cmp_visible_1' if target.type not in ['redshift', 'postgres'] else var('com_snowplowanalytics_snowplow_cmp_visible_1'),
+          'unstruct_event_com_snowplowanalytics_snowplow_consent_preferences_1' if target.type not in ['redshift', 'postgres'] else var('com_snowplowanalytics_snowplow_consent_preferences_1')
+      ],
+      "snowplow__enable_cwv": [
+          'unstruct_event_com_snowplowanalytics_snowplow_web_vitals_1' if target.type not in ['redshift', 'postgres'] else var('snowplow__cwv_events')
+      ],
+      "snowplow__enable_app_errors": [
+          'unstruct_event_com_snowplowanalytics_snowplow_application_error_1' if target.type not in ['redshift', 'postgres'] else var('com_snowplowanalytics_snowplow_application_error_1')
+      ]
+  }
+
+
+  %}
+
+  {{ return(adapter.dispatch('context_existance_check', 'snowplow_unified')(contexts)) }}
+
+{% endmacro %}
+
+{% macro default__context_existance_check(contexts) %}
+
+  {% if execute %}
+    {%- if flags.WHICH in ('run', 'run-operation') -%}
+
+      {% set relation = adapter.get_relation(
+          database=target.database,
+          schema=var('snowplow__atomic_schema', 'atomic'),
+          identifier=var('snowplow__events_table', 'events'))
```

Comment:
We need to use the correct database which might not be the target database.

Reply:
Ok we need to it in way by explicitly stating the db, I am pretty sure we need to define the relation ship with that way

Reply:
Maybe `this.database` ?